### PR TITLE
support custom serializer config

### PR DESCRIPTION
**Pull request by jaspercloud**
### Ⅰ. Describe what this PR did
support custom serializer config.
use interface SerializerConfigurerAdapter.
junit CustomSerializerConfigurerAdapterTest.

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java [hunk 1]
```diff
@@ -150,13 +109,13 @@ public BranchUndoLog decode(byte[] bytes) {
      * if necessary
      * extend {@link ArraySerializerBase}
      */
-    private static class TimestampSerializer extends JsonSerializer<Timestamp> {
+    public static class TimestampSerializer extends JsonSerializer<Timestamp> {
```
Comment (slievrly) [inline rm-datasource/src/main/java/io/seata/rm/datasource/undo/parser/JacksonUndoLogParser.java]: how about remove to `JacksonConfigurerAdapter`?

**Comment by slievrly** [comment]
will conflict with #2445

**Comment by a364176773** [comment]
Resolve code conflicts

**Comment by funky-eyes** [comment]
Due to code conflicts and age, if you are still interested in participating in the community, please resubmit this pr to the 2.x branch after resolving the conflict.
由于代码冲突以及年代久远，如果你还有兴趣参与到社区，请解决完冲突后重新请提交这份pr至2.x分支中

**Pull request closed without merging**